File: java/src/main/java/com/anahoret/anadea_bot/dto/Board.java
package com.anahoret.anadea_bot.dto;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Arrays;

@Getter
@Setter
@NoArgsConstructor
public class Board {

    private int size;
    private int[][] cells;

    @Override
    public String toString() {
        return "Board{" +
                "size=" + size +
                ", cells=" + Arrays.deepToString(cells) +
                '}';
    }
}

File: scala/app/dto/Board.scala
package dto

import java.util
import scala.collection.convert.wrapAll._

import play.api.libs.json.Json

object Board {

  implicit val format = Json.format[Board]

}

case class Board(size: Int, cells: Array[Array[Int]]) {
  override def toString: String =
    "Board{" + "size=" + size +
    ", cells=" + util.Arrays.deepToString(cells.asInstanceOf[Array[AnyRef]])  + "}"

}
